static/app/components/events/autofix/autofixOutputStream.tsx

```diff
@@ -82,50 +87,57 @@ export function AutofixOutputStream({
   useEffect(() => {
     const newText = stream;
+    const previousStream = previousStreamPropRef.current;
+    const separator = '\n\n==========\n\n';
 
-    if (!newText.startsWith(displayedText)) {
-      previousText.current = newText;
-      currentIndexRef.current = 0;
-      setDisplayedText('');
-    }
+    const currentSegmentDisplayed = displayedText.slice(
+      accumulatedTextRef.current.length
+    );
+    const isReset =
+      newText !== previousStream && !newText.startsWith(currentSegmentDisplayed);
 
-    const interval = window.setInterval(() => {
-      if (currentIndexRef.current < newText.length) {
-        setDisplayedText(newText.slice(0, currentIndexRef.current + 1));
-        currentIndexRef.current++;
+    if (isReset) {
+      if (displayedText === previousStream) {
+        accumulatedTextRef.current = displayedText + separator;
+        currentIndexRef.current = accumulatedTextRef.current.length;
       } else {
-        window.clearInterval(interval);
+        const fullPreviousTextWithSeparator = previousStream + separator;
+        setDisplayedText(fullPreviousTextWithSeparator);
+        accumulatedTextRef.current = fullPreviousTextWithSeparator;
+        currentIndexRef.current = fullPreviousTextWithSeparator.length;
       }
-    }, 5);
+    }
 
-    return () => {
-      window.clearInterval(interval);
-    };
-  }, [displayedText, stream]);
+    previousStreamPropRef.current = newText;
 
-  // Animation for active log
-  useEffect(() => {
-    const newActiveLog = activeLog;
+    const combinedText = accumulatedTextRef.current + newText;
 
-    if (!newActiveLog.startsWith(displayedActiveLog)) {
-      previousActiveLog.current = newActiveLog;
-      activeLogIndexRef.current = 0;
-      setDisplayedActiveLog('');
+    if (currentIndexRef.current > combinedText.length) {
+      currentIndexRef.current = combinedText.length;
+      if (displayedText !== combinedText) {
+        setDisplayedText(combinedText);
+      }
     }
 
-    const interval = window.setInterval(() => {
-      if (activeLogIndexRef.current < newActiveLog.length) {
-        setDisplayedActiveLog(newActiveLog.slice(0, activeLogIndexRef.current + 1));
-        activeLogIndexRef.current++;
-      } else {
-        window.clearInterval(interval);
-      }
-    }, 15);
+    let intervalId: number | undefined;
+    if (currentIndexRef.current < combinedText.length) {
+      intervalId = window.setInterval(() => {
+        if (currentIndexRef.current < combinedText.length) {
+          setDisplayedText(combinedText.slice(0, currentIndexRef.current + 1));
+          currentIndexRef.current++;
+        } else {
+          window.clearInterval(intervalId);
+          intervalId = undefined;
+        }
+      }, 1);
+    }
 
     return () => {
-      window.clearInterval(interval);
+      if (intervalId) {
+        window.clearInterval(intervalId);
+      }
     };
-  }, [displayedActiveLog, activeLog]);
+  }, [stream, displayedText]);
```

Comment:
seems fragile to depend on displayedText when we're setting displayedText in here as well...and I wonder if we can use the new typing animation hook here too?

Reply:
I don't think the hook can be re-used here because this is trying to manage and accumulate multiple constantly changing stream texts vs. just animating one static piece of text.

And the dependency on displayedText is needed for the reset logic when combining multiple streams